tests/plugins/lti_consumer/test_forms.py

```diff
@@ -111,3 +120,127 @@ def test_forms_lti_consumer_clean_valid(self):
+        form = LTIConsumerForm(data=data)
+        self.assertFalse(form.is_valid())
+        self.assertDictEqual(
+            form.errors,
+            {
+                "url": [
+                    (
+                        "The url is not valid for this provider. "
+                        'It should be of the form "http://localhost:8060/lti/videos/1234abcd-1".'
+                    )
+                ]
+            },
+        )
+
+    @override_settings(RICHIE_LTI_PROVIDERS=get_lti_settings(is_regex=False))
+    def test_forms_lti_consumer_url_not_regex_included(self):
+        """
+        The url field should include the provider's base url if a predefined LTI provider
+        is used and has a regex url.
+        """
+        data = {"lti_provider_id": "lti_provider_test", "url": "http://invalid.com"}
+        form = LTIConsumerForm(data=data)
+        self.assertFalse(form.is_valid())
+        self.assertDictEqual(
+            form.errors,
+            {
+                "url": [
+                    (
+                        "The url is not valid for this provider. "
+                        'It should start with "http://localhost:8060/lti/videos/".'
+                    )
+                ]
+            },
+        )
+
+    @override_settings(RICHIE_LTI_PROVIDERS=get_lti_settings())
+    def test_forms_lti_consumer_reset_credentials(self):
+        """
+        The "oauth_consumer_key" and "shared_secret" fields should be reset when a value
+        is set for the "lti_provider_id" field.
+        """
+        lti_consumer = LTIConsumerFactory(
+            lti_provider_id=None,
+            oauth_consumer_key="OauthConsumerKey",
+            shared_secret="SharedSecret",
+        )
+        self.assertIsNotNone(lti_consumer.oauth_consumer_key)
+        self.assertIsNotNone(lti_consumer.shared_secret)
+
+        data = LTIConsumerForm(instance=lti_consumer).initial
+        data.update(
+            {
+                "lti_provider_id": "lti_provider_test",
+                "url": "http://localhost:8060/lti/videos/166d465f-f",
+            }
+        )
+        form = LTIConsumerForm(instance=lti_consumer, data=data)
+        form.is_valid()
+        self.assertTrue(form.is_valid())
+
+        form.save()
+        lti_consumer.refresh_from_db()
+        self.assertIsNone(lti_consumer.oauth_consumer_key)
+        self.assertIsNone(lti_consumer.shared_secret)
+
+    @override_settings(RICHIE_LTI_PROVIDERS=get_lti_settings())
+    def test_forms_lti_consumer_shared_secret_placeholder(self):
+        """
+        The "form_shared_secret" should act as a proxy to the "shared_secret" field on the model
+        and allow hiding the shared secret from the form after creation.
+        """
+        lti_consumer = LTIConsumerFactory(
+            lti_provider_id=None,
+            oauth_consumer_key="OauthConsumerKey",
+            shared_secret="SharedSecret",
```

Comment:
```suggestion
            shared_secret="thisIsAtestSharedSecret",
```
I didn't read the test properly at first, maybe other will have the same misunderstanding.
I would just just change the shared_secret to clarify this point 
thisIsAtestSharedSecret